internal/providers/zvm/zvm.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The QEMU provider fetches a local configuration from the firmware config
+// interface (opt/com.coreos/config).
+
+package zvm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/shared/errors"
+	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
+	"github.com/coreos/ignition/v2/internal/log"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+	"github.com/coreos/vcontext/report"
+)
+
+const authorizedSenders string = "*"
+
+func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	/*fetch config files directly from reader device.*/
+	_, err := f.Logger.LogCmd(exec.Command("/sbin/modprobe", "vmur"), "loading zVM control program module")
```

Comment:
nit: paths or names of executables we keep in the `internal/distro` package. We dropped using full paths in favor of just letting them resolve using the `PATH` because different distros put things in different places.

Reply:
nit: I personally like compound `if _, err := fn(); err != nil {` style if's, but this is fine too.

Reply:
yes.